src/gui/N2Go_Gui.php

```diff
@@ -183,50 +195,40 @@ public function getForms($authKey = '')
      * Creates request and returns response. New API
      *
      * @param string $action
-     * @return array
+     * @return array|boolean
      * @internal param mixed $params
      */
-    private function executeNewApi($action)
+    private function executeNewApi($action, $method, $access_token)
     {
-
-        $access_token = get_option('n2go_accessToken');
-
-        $response = wp_remote_get(self::N2GO_API_URL . $action, array(
-                'method' => 'GET',
+        $response = wp_remote_get(
+            self::N2GO_API_URL . $action,
+            array(
+                'method' => $method,
                 'timeout' => 45,
                 'headers' => array('Authorization' => 'Bearer ' . $access_token),
             )
         );
 
-
-        //access_token is deprecated
-        if (is_wp_error($response)) {
-
-            $this->apiErrorMessage =  $response->get_error_message($response->get_error_code());
-            return;
-
-        } else if (isset($response['response']['code']) && $response['response']['code'] == 403 || $response['response']['code'] == 401) {
-
-            $this->refreshTokens();
-            $access_token = get_option('n2go_accessToken');
-            $response = wp_remote_get(self::N2GO_API_URL . $action, array(
-                    'method' => 'GET',
+        if($this->verifyResponse($response)){
+            wp_remote_get(
+                self::N2GO_API_URL . $action,
+                array(
+                    'method' => $method,
```

Comment:
this should not be needed because **wp_remote_get()** makes only GET actions and for POST requests wp_remote_post() can be used.

(https://codex.wordpress.org/Function_Reference/wp_remote_get)